internal/export/exportfile_test.go

```diff
@@ -0,0 +1,158 @@
+		FilenameRoot:     "files",
+		StartTimestamp:   batchStartTime,
+		EndTimestamp:     batchEndTime,
+		Region:           "US",
+		Status:           "",
+		LeaseExpires:     time.Time{},
+		SignatureInfoIDs: []int64{1, 2},
+	}
+
+	exposures := []*publishmodel.Exposure{
+		{
+			ExposureKey:      []byte("ABC"),
+			Regions:          []string{"US"},
+			IntervalNumber:   18,
+			IntervalCount:    0,
+			CreatedAt:        batchStartTime,
+			LocalProvenance:  true,
+			TransmissionRisk: 8,
+		},
+		{
+			ExposureKey:      []byte("DEF"),
+			Regions:          []string{"CA"},
+			IntervalNumber:   118,
+			IntervalCount:    1,
+			CreatedAt:        batchEndTime,
+			LocalProvenance:  true,
+			TransmissionRisk: 1,
+		},
+	}
+
+	signatureInfo := &model.SignatureInfo{
+		SigningKey:        "/kms/project/key/1",
+		SigningKeyVersion: "1",
+		SigningKeyID:      "310",
+		EndTimestamp:      batchEndTime,
+	}
+
+	signer := &customTestSigner{}
+
+	blob, err := MarshalExportFile(batch, exposures, 1, 1, []ExportSigners{
+		{SignatureInfo: signatureInfo,
+			Signer: signer}})
+	if err != nil {
+		t.Fatalf("Can't marshal export file, %v", err)
+	}
+
+	filePath := tmp + "/test-export.zip"
+	err = ioutil.WriteFile(filePath, blob, 0644)
+	if err != nil {
+		t.Fatalf("Can't create an export file: %v", err)
+	}
+
+	blob, err = ioutil.ReadFile(filePath)
+	if err != nil {
+		t.Errorf("Can't read sample file %v: %w", filePath, err)
+	}
+
+	got, err := UnmarshalExportFile(blob)
+	if err != nil {
+		t.Fatalf("Unmarshal failed: %v", err)
+	}
+
+	infos := []*export.SignatureInfo{
+		{
+			VerificationKeyVersion: proto.String("1"),
+			VerificationKeyId:      proto.String("310"),
+			SignatureAlgorithm:     proto.String("1.2.840.10045.4.3.2"),
+		},
+	}
+
+	keys := []*export.TemporaryExposureKey{
+		{
+			KeyData:                    []byte("ABC"),
+			TransmissionRiskLevel:      proto.Int32(8),
+			RollingStartIntervalNumber: proto.Int32(18),
+			RollingPeriod:              proto.Int32(0),
+		},
+		{
+			KeyData:                    []byte("DEF"),
+			TransmissionRiskLevel:      proto.Int32(1),
+			RollingStartIntervalNumber: proto.Int32(118),
+			RollingPeriod:              proto.Int32(1),
+		},
+	}
+
+	keyExport := &export.TemporaryExposureKeyExport{
+		StartTimestamp: proto.Uint64(uint64(batchStartTime.Unix())),
+		EndTimestamp:   proto.Uint64(uint64(batchEndTime.Unix())),
+		Region:         proto.String("US"),
+		BatchNum:       proto.Int32(1),
+		BatchSize:      proto.Int32(1),
+		SignatureInfos: infos,
+		Keys:           keys,
+	}
+
+	ignoredTemporaryExposureKeyExportFields := cmpopts.IgnoreUnexported(export.TemporaryExposureKeyExport{})
+	ignoredSignatureInfoFields := cmpopts.IgnoreUnexported(export.SignatureInfo{})
+	ignoredTemporaryExposureKeyFields := cmpopts.IgnoreUnexported(export.TemporaryExposureKey{})
+	if diff := cmp.Diff(keyExport, got, ignoredTemporaryExposureKeyExportFields, ignoredSignatureInfoFields, ignoredTemporaryExposureKeyFields); diff != "" {
```

Comment:
Minor nit: given how long these fieldnames are perhaps we can put this on its own line so
```go
	diff := cmp.Diff(keyExport, got, ignoredTemporaryExposureKeyExportFields, ignoredSignatureInfoFields, ignoredTemporaryExposureKeyFields)
        if diff != "" {
```